### PR TITLE
docs: mention filtered by ACLs in coordinate API

### PR DESCRIPTION
**Pull request by jkirschner-hashicorp**
I'm concerned that the `X-Consul-Results-Filtered-By-ACLs` response header added in 1.11 (see [API docs](https://www.consul.io/api-docs#results-filtered-by-acls)) has the following UX issues:
- Users are unlikely to be aware of it, unless they happen to have caught this in the changelog or carefully (re-)read the main API docs page
- For users who are aware of it, it may not be obvious which endpoints this response header exists for

This PR is intended to help us align on an approach to these issues (through a docs change) that we can apply across affected HTTP API endpoints. It shows a docs change for a single API (the coordinate API, which has a single affected endpoint).

[**Preview - coordinate endpoint**](https://consul-ot9on7v8r-hashicorp.vercel.app/api-docs/coordinate#read-lan-coordinates-for-all-nodes)
[**Preview - header curl example**](https://consul-ot9on7v8r-hashicorp.vercel.app/api-docs#results-filtered-by-acls)

image

### CODE REVIEW

**Comment by boxofrad** [comment]
I like it! 👏🏻 seems like it might be a bit labor-intensive to add it to all of the affected endpoints though?

**Comment by jkirschner-hashicorp** [comment]
> it might be a bit labor-intensive to add it to all of the affected endpoints though

I think it will take ~45 minutes. The alternative is that users might not be aware of this useful debugging capability, or not be sure where it does and does not apply. I think it's worth it.

File: website/content/api-docs/coordinate.mdx [hunk 1]
```diff
@@ -76,7 +76,8 @@ within the same area.
 ## Read LAN Coordinates for all nodes
 
 This endpoint returns the LAN network coordinates for all nodes in a given
-datacenter.
+datacenter and
+[whether some results were filtered out due to ACL policies](/api-docs#results-filtered-by-acls).
```
[the remark this answers is not in the document]
Comment (jkirschner-hashicorp) [reply website/content/api-docs/coordinate.mdx:80]: @trujillo-adam : I agree that this isn't clear. Two approaches come to mind for how to address this. Let me know whether we should do 1, 2, or both:

1. Reword the link part as. "This endpoint also indicates via the [`X-Consul-Results-Filtered-By-ACLs` HTTP response header](/api-docs#results-filtered-by-acls) whether some results were filtered out due to ACL policies."
2. Add an example to https://www.consul.io/api-docs#results-filtered-by-acls showing how to make a request and find this header in a response using cURL.

[the remark this answers is not in the document]
Comment (jkirschner-hashicorp) [reply website/content/api-docs/coordinate.mdx:80]: Or whether these seem in sufficient / there's a better option you think of.

[the remark this answers is not in the document]
Comment (trujillo-adam) [reply website/content/api-docs/coordinate.mdx:80]: OOOHHH, so the response header could include the `X-Consul-Results-Filtered-By-ACLs` field. Is it a Boolean, like, `Content:X-Consul-Results-Filtered-By-ACLs: true` or just some unassigned string of metadata? I see a few other header fields sprinkled throughout the page, but I don't see like a reference for the response headers that our endpoints could return. To me, documenting that would be the priority because it could provide a scannable single point of reference. 
This is how I would reword that bit we're discussing:
"This endpoint returns the LAN network coordinates for all node in a given datacenter. If ACLs are enabled, the response header may include the `Content:X-Consul-Results-Filtered-By-ACLs: true` parameter, which indicates that the results were filtered according the ACL policy configuration. Refer to the API header reference for additional information."

[the remark this answers is not in the document]
Comment (jkirschner-hashicorp) [reply website/content/api-docs/coordinate.mdx:80]: Regarding the suggestion to have a Consul HTTP API header reference page/section, [I had the same realization while working on this task](https://hashicorp.slack.com/archives/CPEPBFDEJ/p1646261798331069). I intend to create a Github issue for this so it can be worked on at some point. However, I don't think creating that reference (or a starting point for it) is something I have time for right now, so I was planning to keep that out of scope for this PR. Do you agree with that approach (addressing that header reference separately)?

#### Summary

Though I prefer option 1, I'm happy to go with either of the below options. Let me know and I'll proceed. I can define this in a partial so that we can reference the same text everywhere (instead of lots of copy and paste).

##### Option 1
> This endpoint returns the LAN network coordinates for all nodes in a given datacenter.
>
> The [HTTP response header `X-Consul-Results-Filtered-By-ACLs: true`](https://github.com/api-docs#results-filtered-by-acls) will be present if some results were filtered out due to ACL policy configuration.

##### Option 2
> This endpoint returns the LAN network coordinates for all nodes in a given datacenter.
>
> If ACLs are enabled, the HTTP response may include the header `X-Consul-Results-Filtered-By-ACLs: true` indicating that the results were filtered according the ACL policy configuration. Refer to the [`X-Consul-Results-Filtered-By-ACLs` header reference](https://github.com/api-docs#results-filtered-by-acls) for additional information.

#### Detailed Thoughts

**The thoughts below led to the summary above, you can probably skip this if short for time**

> If ACLs are enabled, the response header may include the X-Consul-Results-Filtered-By-ACLs: true parameter, which indicates that the results were filtered according the ACL policy configuration. Refer to the API header reference for additional information.

In my initial wording (`[whether some results were filtered out due to ACL policies](/api-docs#results-filtered-by-acls)`), I tried to keep the language on the endpoint itself concise and use the link to give more info about how to actually use it.  It sounds like as a reader yourself (when reviewing), the wording (though concise) wasn't communicating effectively. I was worried that writing 1-2 standalone sentences would be too much, and also be fully redundant with the link that provides "additional information" (that is only 2 sentences itself): https://www.consul.io/api-docs#results-filtered-by-acls

In an attempt to write something clearer that isn't fully redundant with the "additional info" link, I came up with the below... But the main reason it's shorter than what you suggested is that the link is inline rather than in a "Refer to ______ for additional info".

> This endpoint returns the LAN network coordinates for all nodes in a given datacenter
>
> This endpoint also indicates via the [X-Consul-Results-Filtered-By-ACLs HTTP response header](https://github.com/api-docs#results-filtered-by-acls) whether some results were filtered out due to ACL policies.

[the remark this answers is not in the document]
Comment (trujillo-adam) [reply website/content/api-docs/coordinate.mdx:80]: It takes more space, but I prefer option 2 because it can be more helpful to give readers a sense of the topic you're sending them to. The issue with the style in option 1 (in which we link a phrase or keyword) is that the context of the reference may not match the context of linked phrase/keyword. It's not always the case, but when the linked phrase isn't necessarily the core message of the linked content, then it might be jarring for readers. 

Linked phrases/keywords may also affect the UX. In this instance, we've already provided the important information--the link is supplementary and may actually distract readers from accomplishing their current task.  

But we could be overthinking it.

[the remark this answers is not in the document]
Comment (jkirschner-hashicorp) [reply website/content/api-docs/coordinate.mdx]: The text now says:
> The HTTP response header `X-Consul-Results-Filtered-By-ACLs: true` will be present if some results were filtered out due to ACL policy configuration. Refer to the [HTTP API documentation](https://consul-xuisg6nod-hashicorp.vercel.app/api-docs#results-filtered-by-acls) for more information.

**Comment by jkirschner-hashicorp** [comment]
@trujillo-adam : ready for re-review

**Comment by jkirschner-hashicorp** [comment]
@trujillo-adam : just added an example of how to view the response header using `curl`. Once this is approved/merged, I can apply this information to all relevant endpoints easily (using the partial, so the text is easy to maintain).

File: website/content/api-docs/index.mdx [hunk 2]
```diff
@@ -106,6 +106,27 @@ of results, because some have been filtered out by ACL policies.
 In order to limit information leakage, this header is only present for requests
 authenticated by a valid ACL token.
 
+Below is an example using `curl` to view the
+`X-Consul-Results-Filtered-By-ACLs` response header.
```
Comment (trujillo-adam) [inline website/content/api-docs/index.mdx:110]:
```suggestion
The following example uses `curl` to view the
`X-Consul-Results-Filtered-By-ACLs` response header.
```
It's not in the style guide yet, but we want to provide guidance on directional language so that we are consistently describing the order of things. Basically, we want to standardize on "The following . . .", "In the following . . ." as opposed to using "below", "above", "next", "to the right", etc. The idea is to maintain a consistent flow.

File: website/content/partials/http_api_results_filtered_by_acls.mdx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+The HTTP response header `X-Consul-Results-Filtered-By-ACLs: true` will be present
+if some results were filtered out due to ACL policy configuration.
+Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
```
Comment (trujillo-adam) [inline website/content/partials/http_api_results_filtered_by_acls.mdx:3]:
```suggestion
The `X-Consul-Results-Filtered-By-ACLs: true` HTTP header appears in the response if a subset of the results have been filtered out due to ACL policy configuration.
Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
```
I think we want to use the simple present tense as much as possible. The past perfect is good for describing why something turned out the way it did.

**Review by trujillo-adam — COMMENTED** [review]
A few suggestions

File: website/content/partials/http_api_results_filtered_by_acls.mdx (hunk 3 above)
Comment (jkirschner-hashicorp) [reply website/content/partials/http_api_results_filtered_by_acls.mdx:3]: Or:
```
The HTTP response includes the `X-Consul-Results-Filtered-By-ACLs: true` header
if the response array excludes any results due to ACL policy configuration.
Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
```

Comment (jkirschner-hashicorp) [reply website/content/partials/http_api_results_filtered_by_acls.mdx]: I think I prefer the second alternative I gave because it explains it differently than the header name, giving the user two ways of explaining / understanding the same thing. "Results filtered by ACLs" and "response array excludes results".

I think having the header name say "results filtered by ACLs" and then saying "array filtered by ACLs" is just restating the header name without making it any easier to understand.

Comment (trujillo-adam) [reply website/content/partials/http_api_results_filtered_by_acls.mdx]:
```
The HTTP response includes the `X-Consul-Results-Filtered-By-ACLs: true` header
if the response array excludes results that as a result of an ACL policy configuration.
Refer to the [HTTP API documentation](/api-docs#results-filtered-by-acls) for more information.
```
Really starting to split hairs here, but I agree with the idea that the response is an _array_ and that some of the results were _excluded_ as a result of an ACL policy configuration linked with the token.

Comment (jkirschner-hashicorp) [reply website/content/partials/http_api_results_filtered_by_acls.mdx]: ### On "any"
I agree that the word "any" is extraneous, so I'll remove it.

### On "due to" versus "as a result of"
I still think
> excludes results due to ACL policy configuration

is better than
> excludes results as a result of an ACL policy configuration

because we avoid repetition of the word "result".

### Next steps

If I remove the word "any", is this ready to be marked approved? I can't actually merge because you still have an outstanding "changes requested" review.

**Review by trujillo-adam — APPROVED** [review]

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/609291.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit bea07bd6e76f47adead4defa91fec19774fe47c1 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit bea07bd6e76f47adead4defa91fec19774fe47c1 onto `release/1.11.x` succeeded!
